lib/ansible/modules/cloud/vmware/vmware_host_ntp.py

```diff
@@ -0,0 +1,215 @@
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        self.ntp_servers = self.params.get('ntp_servers', list())
+        self.hosts = []
+        if cluster_name:
+            cluster_obj = self.find_cluster_by_name(cluster_name=cluster_name)
+            if cluster_obj:
+                self.hosts = [host for host in cluster_obj.host]
+            else:
+                module.fail_json(changed=False, msg="Cluster '%s' not found" % cluster_name)
+        elif esxi_host_name:
+            esxi_host_obj = self.find_hostsystem_by_name(host_name=esxi_host_name)
+            if esxi_host_obj:
+                self.hosts = [esxi_host_obj]
+            else:
+                module.fail_json(changed=False, msg="ESXi '%s' not found" % esxi_host_name)
+        self.results = {}
+        self.desired_state = module.params['state']
+
+    def update_ntp_servers(self, host, ntp_servers, operation='add'):
+        changed = False
+        host_date_time_manager = host.configManager.dateTimeSystem
+        if host_date_time_manager:
+            available_ntp_servers = host_date_time_manager.dateTimeInfo.ntpConfig.server
+            if operation == 'add':
+                available_ntp_servers = available_ntp_servers + ntp_servers
+            elif operation == 'delete':
+                for server in ntp_servers:
+                    if server in available_ntp_servers:
+                        available_ntp_servers.remove(server)
+
+            ntp_config_spec = vim.host.NtpConfig()
+            ntp_config_spec.server = available_ntp_servers
+            date_config_spec = vim.host.DateTimeConfig()
+            date_config_spec.ntpConfig = ntp_config_spec
+            try:
+                host_date_time_manager.UpdateDateTimeConfig(date_config_spec)
+                self.results[host.name]['after_change_ntp_servers'] = host_date_time_manager.dateTimeInfo.ntpConfig.server
+                changed = True
+            except vim.fault.HostConfigFault as e:
+                self.results[host.name]['error'] = e.msg
+            except Exception as e:
+                self.results[host.name]['error'] = e
+
+        return changed
+
+    def check_host_state(self):
+        change_list = []
+        changed = False
+        for host in self.hosts:
+            ntp_servers_to_change = self.check_ntp_servers(host=host)
+            self.results[host.name].update(dict(
+                ntp_servers_to_change=ntp_servers_to_change,
+                desired_state=self.desired_state,
+            )
+            )
+
+            if not ntp_servers_to_change:
+                change_list.append(False)
+                self.results[host.name]['current_state'] = self.desired_state
+                continue
+            elif ntp_servers_to_change:
+                if self.desired_state == 'present':
+                    changed = self.update_ntp_settings(host=host, ntp_servers=ntp_servers_to_change)
```

Comment:
update_ntp_settings → update_ntp_servers (or re-change the update_ntp_servers method name)